cuda/base/exception.cpp

```diff
@@ -98,6 +101,29 @@ std::string CusparseError::get_error(int64 error_code)
 }
 
 
+#if GKO_HAVE_LAPACK
```

Comment:
need to provide the else part. the device_hook is not compiled when compiling cuda without lapack.

Reply:
I was thinking that we should have all the other code guarded such that we will never try to call `CusolverError::get_error` anywhere if building without LAPACK (and will get errors if we do, so we will immediately realize we have code that should be guarded) so would we still need the else in that case?

Reply:
you have the declaration of CusolverError, so you need to implement it somewhere. Do you guard the cusolverError with this Macro? maybe I miss it.

Reply:
No, I didn't have the definition guarded, just where it was used. I will add the else.